Sources/JXKit/JXContext.swift

```diff
@@ -4,24 +4,63 @@ import JavaScriptCore
 #else
 import CJSCore
 #endif
-#if canImport(FoundationNetworking)
-import FoundationNetworking
+#if canImport(Foundation)
+import Foundation
 #endif
 
 /// A JavaScript execution context. This is a cross-platform analogue to the Objective-C `JavaScriptCore.JSContext`.
 ///
 /// This wraps a `JSGlobalContextRef`, and is the equivalent of `JavaScriptCore.JSContext`
 open class JXContext {
+    /// Context configuration.
+    public struct Configuration {
+        /// Whether scripts evaluated by this context should be assessed in `strict` mode. Defaults to `true`.
+        public var strict: Bool
+        
+        /// Whether hot reloading of JavaScript script resources is enabled.
+        public var isHotReloadEnabled: Bool {
+#if canImport(Foundation)
+            return scriptLoader.didChange != nil
+#else
+            return false
+#endif
+        }
+
+#if canImport(Foundation)
+        /// Whether `require` module support is enabled. Defaults to `true`.
+        public var moduleRequireEnabled: Bool
+        
+        /// Configure a global script loader to use as the default when no loader is provided to the `Configuration`.
+        ///
+        /// - Seealso: ``JXContext/Configuration/scriptLoader``
+        public static var defaultScriptLoader: JXScriptLoader = DefaultScriptLoader()
```

Comment:
Right now this is the very simple mechanism that would allow another package to inject more advanced script loading and monitoring globally, without the developer having to pass a custom loader every time they create a context.